app/master/build.py

```diff
@@ -564,6 +569,121 @@ def _generate_unique_symlink_path_for_build_repo(self):
         """
         return os.path.join(Configuration['build_symlink_directory'], str(uuid.uuid4()))
 
+    def save(self):
```

Comment:
`save` assumes that this build already exists in the database and we're just updating its values.

`_store_build` is creating this build and storing it for the first time.

They both work slightly differently -- `save` queries the database using SQLAlchemy for the model and `_store_build` creates the model from the build. Maybe there's a way to condense these?